### DISCUSSION TITLE
v3.7.0

### DISCUSSION THREAD

**Discussion by krashish8** [pgRouting/pgrouting#2677] [Announcements]
# pgRouting 3.7.0 Release Notes

To see all issues & pull requests closed by this release see the [Git closed milestone for 3.7.0](https://github.com/pgRouting/pgrouting/issues?utf8=%E2%9C%93&q=milestone%3A%22Release%203.7.0%22)

**Support**

* [#2656](https://github.com/pgRouting/pgrouting/pull/2656) Stop support of PostgreSQL12 on pgrouting v3.7

  * Stopping support of PostgreSQL 12
  * CI does not test for PostgreSQL 12

**New experimental functions**

* Metrics

  * pgr_betweennessCentrality

**Official functions changes**

* [#2605](https://github.com/pgRouting/pgrouting/pull/2605) Standarize spanning tree functions output

  * Functions:

    * ``pgr_kruskalDD``
    * ``pgr_kruskalDFS``
    * ``pgr_kruskalBFS``
    * ``pgr_primDD``
    * ``pgr_primDFS``
    * ``pgr_primBFS``

  * Standarizing output columns to ``(seq, depth, start_vid, pred, node, edge, cost, agg_cost)``
    * Added ``pred`` result columns.

**Experimental promoted to proposed.**

* [#2635](https://github.com/pgRouting/pgrouting/pull/2635) pgr_LineGraph ignores directed flag and use negative values for identifiers.

  * ``pgr_lineGraph``

    * Promoted to **proposed** signature.
    * Works for directed and undirected graphs.

**Code enhancement**

* [#2599](https://github.com/pgRouting/pgrouting/pull/2599) Driving distance cleanup
* [#2607](https://github.com/pgRouting/pgrouting/pull/2607) Read postgresql data on C++
* [#2614](https://github.com/pgRouting/pgrouting/pull/2614) Clang tidy does not work

### 
**Attachments**
File | Contents
| --- | --- |
| `doc-v3.7.0.tar.gz` | English and Spanish documentation. Redirection to English
| `pgrouting-3.7.0.tar.gz` | tar.gz of the release
| `pgrouting-3.7.0.zip`| zip of the release

<hr /><em>This discussion was created from the release <a href='https://github.com/pgRouting/pgrouting/releases/tag/v3.7.0'>v3.7.0</a>.</em>

**Comment by krashish8** [comment]
## What's Changed
* Creating new minor 3.7.0 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2597
* Update contributors by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2598
* Drivingdistance cleanup by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2599
* Test chinese by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2600
* Update notes2news script and applying it by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2601
* Fixing issue 2602 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2604
* Deprecating min supported versions by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2603
* Standarizing output spanning tree functions by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2606
* Read postgresql data on C++ by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2607
* Update mailing list to discourse by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2610
* (C++)(lint) adding mising includes after code checkr works by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2613
* Fix Issue 2614 move h to hpp by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2615
* adding a check for clang-tidy by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2620
* Fixing some problems detected with clang-tidy by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2621
* Remove seq from path rt by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2622
* Use bool instead of enum on graph types by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2623
* Cleanup pgr_base_graph  by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2624
* Add macos-14 to macos CI with fixing build error by @sanak in https://github.com/pgRouting/pgrouting/pull/2626
* Update release script to use NEWS.md on develop by @krashish8 in https://github.com/pgRouting/pgrouting/pull/2631
* Add 3.6.2 info to 3.7.0 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2633
* Remove bool from definition of a graph by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2634
* Fix pgr line graph by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2625
* Translations update from OSGeo Weblate by @osgeoweblate in https://github.com/pgRouting/pgrouting/pull/2592
* Modernize doc queries generator by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2637
* Some code enhancements by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2642
* Fix new lint errors by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2640
* New function pgr betweenness centrality by @bedupako12mas in https://github.com/pgRouting/pgrouting/pull/2647
* Fix 2648 fail building under window mingw64 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2649
* Translations update from OSGeo Weblate by @osgeoweblate in https://github.com/pgRouting/pgrouting/pull/2645
* Fixing some errors detected on weblate by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2651
* Translations update from OSGeo Weblate by @osgeoweblate in https://github.com/pgRouting/pgrouting/pull/2650
* Some manual changes on chinese documentation by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2652
* Fix sphinx errors on chinese by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2653
* Fix chinese documentation by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2654
* Recover macos test by @sanak in https://github.com/pgRouting/pgrouting/pull/2638
* Issue 2656 stop support of postgre sql12 on pgrouting v3.7 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2657
* Port main 3.6.3 changes to develop 3.7.0 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2660
* Translations update from OSGeo Weblate by @osgeoweblate in https://github.com/pgRouting/pgrouting/pull/2661
* Bump softprops/action-gh-release from 1 to 2 by @dependabot in https://github.com/pgRouting/pgrouting/pull/2666
* Port fixes from master by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2665
* Remove pgr prefix from files by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2669
* (CI) Creating an update path from 3.6.3 to 3.7.0 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2671
* (tool) remove trailing space on NEWS.md by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2672
* Translations update from OSGeo Weblate by @osgeoweblate in https://github.com/pgRouting/pgrouting/pull/2668
* Remove -dev before PR to main by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2673
* Resolve pr merge conflicts develop with main by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2675
* Preparing for release 3.7.0 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2674
* Fix requires by @rdunklau in https://github.com/pgRouting/pgrouting/pull/2676

## New Contributors
* @rdunklau made their first contribution in https://github.com/pgRouting/pgrouting/pull/2676

**Full Changelog**: https://github.com/pgRouting/pgrouting/compare/v3.6.3...v3.7.0

**Comment by stevengardiner** [comment]
Hi Do you know when pgrouting will be available to postgresql 16 in an azure flexible server?
